sentinel/pkg/checker/event/app.go

```diff
@@ -121,20 +121,36 @@ func checkFeeds(ctx context.Context, FeedsToCheck []FeedToCheck) {
 	msg := ""
 	for i, feed := range FeedsToCheck {
 		offset, err := timeSinceLastFeedEvent(ctx, feed)
-		if err != nil {
+		if err == nil {
+			if offset > time.Duration(feed.ExpectedInterval)*time.Millisecond*2 {
+				log.Warn().Str("feed", feed.FeedName).Msg(fmt.Sprintf("%s delayed by %s", feed.FeedName, offset-time.Duration(feed.ExpectedInterval)*time.Millisecond))
+				FeedsToCheck[i].LatencyChecked++
+				if FeedsToCheck[i].LatencyChecked > AlarmOffset {
+					msg += fmt.Sprintf("%s delayed by %s\n", feed.FeedName, offset-time.Duration(feed.ExpectedInterval)*time.Millisecond)
+					FeedsToCheck[i].LatencyChecked = 0
+				}
+			} else {
+				FeedsToCheck[i].LatencyChecked = 0
+			}
+		} else {
```

Comment:
I think the depth is too deep here, lets encapsulate inner block into separate function. it'd be good to go same for max feed submission checks

Reply:
good point, done!

Reply:
- `msg` is passed as a pointer
- build is successful as well